src/cdk/generator.ts

```diff
@@ -78,9 +87,13 @@ export class ResourceGenerator extends Construct {
     let apiViewConstruct: undefined | ApiViewConstruct
     if (apiViewMeta) {
       const name = apiViewMeta.apiClass.name
+
+      // merge defaults with config in meta
+      const mergedOptions: Omit<ApiProps, "httpApi"> = deepmerge(this.functionOptions || {}, apiViewMeta)
```

Comment:
Sorry, I don't understand what this achieves and why

Reply:
I updated the comment hopefully it's clearer